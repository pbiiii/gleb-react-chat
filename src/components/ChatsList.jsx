import React from 'react';
import PropTypes from 'prop-types';
import List from '@material-ui/core/List';
import { withStyles } from '@material-ui/core/styles/index';
import { Typography } from '@material-ui/core';
import ChatItem from './ChatItem';
import ChatType from '../types/ChatType';

const styles = () => ({
    subtitle: {
        textAlign: 'center',
    },
    list: {
        height: '100%',
        overflowY: 'scroll',
    },
});

const ChatsList = ({
    chats, classes, activeChat, disabled,
}) => (
    <List className={classes.list}>
        {chats && chats.length > 0 ? (
            chats.map(chat => (
                <ChatItem
                    disabled={disabled}
                    chat={chat}
                    key={chat._id}
                    active={Boolean(activeChat && activeChat._id === chat._id)}
                />
            ))
        ) : (
            <Typography variant="body1" className={classes.subtitle}>
                There is no chats yet...
            </Typography>
        )}
    </List>
);

ChatsList.propTypes = {
    classes: PropTypes.objectOf(PropTypes.string).isRequired,
    activeChat: PropTypes.shape(ChatType),
    chats: PropTypes.arrayOf(PropTypes.shape(ChatType)).isRequired,
    disabled: PropTypes.bool.isRequired,
};

ChatsList.defaultProps = {
    activeChat: null,
};

export default withStyles(styles)(ChatsList);
