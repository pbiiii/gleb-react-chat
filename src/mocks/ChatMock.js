import UserMock from './UserMock';

export default {
    _id: 'ashasdaksdmkl1mkl',
    creator: { ...UserMock, _id: 2 },
    title: 'Mock title',
};
