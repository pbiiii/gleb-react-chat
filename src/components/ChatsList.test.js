/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatsList from './ChatsList';
import ChatMock from '../mocks/ChatMock';

const mockProps = {
    activeChat: { ...ChatMock },
    chats: [ChatMock],
    disabled: false,
};

jest.mock('./ChatItem', () => () => 'ChatItem');

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatsList {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('render correctly', () => {
    const tree = renderer
        .create(<ChatsList {...mockProps} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
