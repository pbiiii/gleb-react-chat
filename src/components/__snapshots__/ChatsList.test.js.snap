// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<ul
  className="MuiList-root-9 MuiList-padding-10 ChatsList-list-8"
>
  ChatItem
</ul>
`;
